lib/cocoapods-fix-react-native/fix_with_context.rb

```diff
@@ -0,0 +1,92 @@
+class CocoaPodsFixReactNative
+  def post_fix_with_context(context)
+    # Get the current version of React Native in your app
+    react_spec = context.sandbox.specification('React')
+    if react_spec.nil?
+      Pod::UI.warn "No React dependency found"
+      return
+    end
+
+    version = react_spec.version.to_s
+
+    unless patch_exists?(version)
+      Pod::UI.warn "CP-Fix-React-Native does not support #{version} yet, please send " +
+                   'PRs to https://github.com/orta/cocoapods-fix-react-native'
+      return
+    end
+
+    path_to_fix = version_file(version)
+
+    if File.exist? path_to_fix
+      Pod::UI.section "Patching React Native #{version}" do
+        require(path_to_fix)
+      end
+    end
+  end
+
+  def pre_fix_with_context(context)
+    # Get the current version of React Native in your app
+    require 'molinillo/dependency_graph'
+
+    locked_dependencies = Molinillo::DependencyGraph.new
+    context.lockfile.dependencies.each do |dependency|
+      locked_dependencies.add_vertex(dependency.name, dependency, true)
+    end
+
+    # TODO resolve raw sources from context.podfile.sources to local sources
+    # sources = context.podfile.sources.map { |s| Pod::Source.new(s) }
+    sources = [Pod::Source.new('~/.cocoapods/repos/uw/'), Pod::Source.new('~/.cocoapods/repos/master/')]
```

Comment:
For our use case I think it's fine to include all repos in `~/.cocoapods/repos/` so maybe `Dir.glob("~/.cocoapods/repos/*").map(Pod::Source.new)`  or similar?

Reply:
I actually managed to transform the sources from the Podfile into `Pod::Source` objects by using  

```rb
Pod::Config.instance.sources_manager.source_with_name_or_url(source)
```